### DISCUSSION TITLE
OSPFv3 over XFRM ipsec interface

### DISCUSSION THREAD

**Discussion by ptchuba** [strongswan/strongswan#1857] [Q&A]
Hello,

I have a tunnel between 2 strongswan hosts using xfrm interfaces and traffic selectors as below:

```
connections {
   hub {
      remote_addrs = 2600::XXXX
      local_addrs = 2600::YYYY
      local {
         auth = psk
         id = hosta
      }
      remote {
         auth = psk
         id = hostb
      }
      children {
         hostb-sa {
            local_ts  = fd01:2:1::/48
            remote_ts = fd01:2::/32
            if_id_in = 1
            if_id_out = 1
...
```

I have the ipsec interface with id 1 configured as an xfrm interface and have assigned an IPv6 ULA address.
output of `ip a`
```
3: ipsec1@lo: <NOARP,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000
    link/none 
    inet6 fd01:2:1:1::2/64 scope global 
       valid_lft forever preferred_lft forever
    inet6 fe80::cd34:5c55:f4d6:d993/64 scope link stable-privacy 
       valid_lft forever preferred_lft forever
```
I've installed frr and enabled ospf6d and configured it on the ipsec1 interfaces of both hosts. My OSPF neighbors don't come up.  So I added `fe80::/64` to local_ts and `ff02::5/128,ff02::6/128,fe80::/64` to remote_ts, seeing as the OSPF messages are sourced from the link local addresses to the allspfrouters and allDRouters addresses ff02::5 and ff02::6 respectively. This still does not work. The `ip xfrm policy` does not contain the ff02 addresses nor are they seen in the child SA in `ipsec statusall` : 
```
...
         hostb[1]: ESTABLISHED 7 seconds ago, 2600::XXXX[hosta]...2600::YYYY[hostb]
         hostb[1]: IKEv2 SPIs: 39b0b5831887310e_i* e8d8676fcd84e6bd_r, rekeying in 3 hours
         hostb[1]: IKE proposal: AES_GCM_16_256/PRF_HMAC_SHA2_512/ECP_384
      hostb-sa{1}:  INSTALLED, TUNNEL, reqid 1, ESP in UDP SPIs: c2b69cbf_i cb7aa5a9_o
      hostb-sa{1}:  AES_GCM_16_256, 2712 bytes_i (11 pkts, 0s ago), 2814 bytes_o (11 pkts, 0s ago), rekeying in 53 minutes
      hostb-sa{1}:   fd01:2:1::/48 fe80::/64 === fd01:2::/32 fe80::/64

...
```

However, when I change both selectors to `::/0`, my neighbors come up and OSPF is happy. What am I missing in the traffic selectors to make this work without using ::/0. I will like to have more control and keep the selectors specific.

Thanks!!

**Comment by ptchuba** [comment]
Any suggestions?

**Comment by tobiasbrunner** [comment]
> The `ip xfrm policy` does not contain the ff02 addresses nor are they seen in the child SA in `ipsec statusall` :

Sounds like you configured the traffic selectors incorrectly (check the logs, possibly with _cfg_ on level 2).

**Comment by ptchuba** [comment]
Thanks @tobiasbrunner . I ended up just using ::/0 selectors. Will return to this some other time.
